src/site/content/en/blog/kapwing/index.md

```diff
@@ -0,0 +1,415 @@
+owners all over the world need quick and easy-to-use tools to make video
+content. 
+
+Companies like Kapwing make it possible to create all this video content right
+on the web, by using the latest in powerful APIs and performance tools. 
+
+## About Kapwing
+
+Kapwing is a web-based collaborative video editor designed mainly for casual
+creatives like game-streamers, musicians, YouTube creators, and meme-rs. It's
+also a go-to resource for business owners who need an easy way to produce
+their own social content, such as Facebook and Instagram ads.
+
+People discover Kapwing by searching for a specific task, for example "how to
+trim a video," "add music to my video," or "resize a video." They can do what
+they searched for with just one click&mdash;without the added friction of
+navigating to an app store and downloading an app. The web makes it simple for
+people to search for precisely what task they need help with, and then do it.
+
+After that first click, Kapwing users can do a whole lot more just as easily.
+They can explore free templates; add new layers of free stock videos, insert
+subtitles, transcribe videos, and upload background music. 
+
+## How Kapwing brings real-time editing and collaboration to the web
+
+While the web provides unique advantages, it also presents distinct
+challenges. Kapwing needs to deliver smooth and precise playback of complex,
+multi-layered projects across a wide range of devices and network conditions.
+
+To this end we use a variety of web APIs to achieve our performance and
+feature goals.
+
+### IndexedDB
+
+High performance editing requires that all of our users' content live on the
+client, avoiding the network whenever possible. Unlike a streaming service,
+where users typically access a piece of content once, our customers reuse
+their assets frequently, days and even months after upload.
+
+IndexedDB allows us to provide persistent file system-like storage to our
+users. The result is that over 90% of media requests in the app are fulfilled
+locally. Integrating indexedDB into our system was very straightforward.
+
+Here is some boiler plate initialization code that runs on app load:
+
+```js
+import { DBSchema, openDB, deleteDB, IDBPDatabase } from "idb";
+let openIdb: Promise<IDBPDatabase<Schema>>;
+const db =
+  (await openDB) <
+  Schema >
+  ("kapwing",
+  version,
+  {
+    upgrade(db, oldVersion) {
+      if (oldVersion >= 1) {
+        // assets store schema changed, need to recreate
+        db.deleteObjectStore("assets");
+      }   
+      db.createObjectStore("assets", { keyPath: "mediaLibraryID" }); 
+    },  
+    async blocked() {
+      await deleteDB("kapwing");
+    },  
+    async blocking() {
+      await deleteDB("kapwing");
+  },
+});
+```
+
+We pass a version and define an upgrade function. This is used for
+initialization or to update our schema when necessary.
+
+We also pass error handling callbacks blocked and blocking, which we have
```

Comment:
> We also pass error handling callbacks blocked and blocking

This doesn't scan for me, and doesn't seem to relate to anything I can see in the code. This either tells me this detail isn't important from a reader's perspective, or that it could use a bit of shoring up.

Reply:
You could raise this question with the Kapwing developer in Google Docs, @heyawhite can share. Keep in mind that we are looking to get this out the door as soon as possible and weigh whether it is worth the discussion.

Reply:
If we're on a tight timeline, then it's not worth it. It could just be me that's having trouble understanding 😅

Reply:
I've clarified this, it is indeed related to the code:

> We pass error-handling callbacks, `blocked` and `blocking`, which we've found useful in preventing issues for users with unstable systems.

Thanks for pointing it out.